roles/openshift_efk/tasks/main.yaml

```diff
@@ -0,0 +1,179 @@
+---
+
+  - fail: msg="This role requires the following vars to be defined: master_url, kibana_hostname, es_cluster_size"
+    when: "kibana_hostname is not defined or
+          es_cluster_size is not defined or
+          master_url is not defined"
+
+  - name: "Checking for logging project"
+    command: oc get project logging
+    register: logging_project
+    failed_when: "'FAILED' in logging_project.stderr"
+    tags:
+      - cleanup
```

Comment:
I can see where you are going with the cleanup/build tags, however it would cause a lot of pain to consume this role with the tags being configured in this way.

Instead I would suggest the following:
- in tasks/main.yaml
  - test that the required variables exist.
  - add an optional variable for openshift_hosted_logging_cleanup
  - conditionally include a deploy/cleanup task file based on the value of the above variable:
```
- name: Cleanup logging deployment
  include: cleanup.yml
  when: openshift_hosted_logging_cleanup | default(false) | bool

- name: Deploy logging
  include: deploy.yml
  when: not openshift_hosted_logging_cleanup | default(false) | bool
```

This would still give you both behaviors, you would just need to create a playbook to get the same tagging functionality back:
```
- hosts: my_master_host
  tags:
  - build
  roles:
  - role: openshift_hosted_logging

- hosts: my_master_host
  tags:
  - cleanup
  roles:
  - role: openshift_hosted_logging
    openshift_hosted_logging_cleanup: yes
```